spec/app/tournament/tournament.spec.js

```diff
@@ -0,0 +1,121 @@
+        expect( tournament.existsPlayer(spiderman) ).toBeUndefined();
+    });
+
+    it('should return a validation error when validating a player who is already enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(spiderman);
+
+        expect( validationResult.error ).toEqual("Player already enrolled");
+    });
+
+    it('should return an unsuccessful validation when validating a player who is already enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(spiderman);
+
+        expect( validationResult.success ).toBe(false);
+    });
+
+    it('should not return any validation error when validating a player who is not yet enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(batman);
+
+        expect( validationResult.error ).toEqual("");
+    });
+
+    it('should return a successful validation when validating a player who is not yet enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(batman);
+
+        expect( validationResult.success ).toBe(true);
+    });
+
+    it('should create 1 match generating with 2 players', function() {
+        tournament.addPlayer(spiderman);
+        tournament.addPlayer(batman);
+        tournament.generate();
+
+        expect( tournament.matches.length ).toEqual(1);
+    });
+
+    it('should create 3 match generating with 4 players', function() {
+        setMockedTournament(4)
+        tournament.generate();
+
+        expect( tournament.matches.length ).toEqual(3);
+    });
+
+    it('should order its players by rank after adding a new player', function() {
+        tournament.addPlayer(theThing);
+
+        expect( tournament.players[0].rank ).toEqual(5);
+
+        tournament.addPlayer(batman);
+
+        expect( tournament.players[0].rank ).toEqual(3);
+        expect( tournament.players[1].rank ).toEqual(5);
+
+        tournament.addPlayer(spiderman);
+
+        expect( tournament.players[0].rank ).toEqual(1);
+        expect( tournament.players[1].rank ).toEqual(3);
+        expect( tournament.players[2].rank ).toEqual(5);
+    });
+
+    it('should have 3 rounds if it has 8 players', function() {
+        setMockedTournament(8);
+        tournament.generate();
+
+        expect( tournament.getRounds() ).toEqual(3);
+    });
+
+    it('should have two round 1 matches with a round 2 child if it has 4 players', function() {
+        setMockedTournament(4);
+        tournament.generate();
+
+        expect( tournament.matches[0].round ).toEqual(1);
+        expect( tournament.matches[0].nextMatch.round ).toEqual(2);
+        expect( tournament.matches[1].round ).toEqual(1);
+        expect( tournament.matches[1].nextMatch.round ).toEqual(2);
```

Comment:
I'm not confortable with matches[0] and matches[1]. I don't think that's part of our spec but just an implementation detail. We could change algorithm and matches get sorted differently, and that would break the test without reason. Verify only what needs to be verified.